pkg/resource/deploy/step.go

```diff
@@ -252,6 +252,12 @@ func (s *CreateStep) Apply(preview bool) (resource.Status, StepCompleteFunc, err
 		s.new.Outputs = outs
 	}
 
+	// If we've created a new resource as a replacement for an old one before we've deleted the old one check
+	// that the ID is a new ID.
+	if s.replacing && s.pendingDelete && s.new.Custom && s.new.ID == s.old.ID {
+		return resourceStatus, nil, fmt.Errorf("provider returned existing ID (%s) for replacement resource", s.new.ID)
```

Comment:
I'm very uncomfortable with this error. From a user's point of view, I can't read it any other way than "Pulumi got really confused about what it's doing and decided to blow up". I suspect we need to improve the workflow in some other way that would be net-beneficial for users and not just bail.

Reply:
I think the choices are:
1) Improve the error to suggest setting [DeleteBeforeReplace](https://www.pulumi.com/docs/intro/concepts/resources/options/deletebeforereplace/).
2) Try the idea above about accepting the create and eliding the provider delete for the "old" resource.